Content.Server/GameObjects/Components/BaseComputerUserInterfaceComponent.cs

```diff
@@ -0,0 +1,144 @@
+using Content.Server.GameObjects.Components.Construction;
+using Content.Server.GameObjects.Components.Power.ApcNetComponents;
+using Content.Server.Utility;
+using Content.Shared.GameObjects.Components;
+using Content.Shared.GameObjects.EntitySystems;
+using Content.Shared.GameObjects.EntitySystems.ActionBlocker;
+using Content.Shared.Interfaces;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.ViewVariables;
+using Robust.Shared.Localization;
+
+namespace Content.Server.GameObjects.Components
+{
+    /// <summary>
+    /// This component is used as a base class for classes like SolarControlConsoleComponent.
+    /// These components operate the server-side logic for the "primary UI" of a computer.
+    /// That means showing the UI when a user activates it, for example.
+    /// </summary>
+    public abstract class BaseComputerUserInterfaceComponent : Component
+    {
+        protected readonly object UserInterfaceKey;
+
+        [ViewVariables] protected BoundUserInterface? UserInterface => Owner.GetUIOrNull(UserInterfaceKey);
+        [ViewVariables] public bool Powered => !Owner.TryGetComponent(out PowerReceiverComponent? receiver) || receiver.Powered;
+
+        public BaseComputerUserInterfaceComponent(object key)
+        {
+            UserInterfaceKey = key;
+        }
```

Comment:
Don't use constructors for components.

Reply:
Constructor *required* due to inheritance pattern, not removable